src/main/java/org/mule/extension/http/api/notification/HttpRequestData.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api.notification;
+
+import org.mule.runtime.api.util.MultiMap;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+
+import java.net.URI;
+
+/**
+ * HTTP notification data regarding requests.
+ *
+ * @since 1.1
+ */
+public class HttpRequestData extends HttpData {
+
+  private final String method;
+  private final URI uri;
```

Comment:
In 3.x it was possible to declare notification listeners on the xml level. If we ever want to do that with Mule 4, is URI a data type that will play nicely?